source/rpt/rpt_cell_reconstruction.cc

```diff
@@ -0,0 +1,626 @@
+      std::cout << "Number of vertices with calculated counts : "
+                << map_vertices_index.size() << std::endl;
+    }
+
+  // Generate visualisation files
+  visualize_positions();
+
+  // Disable the output of time clock
+  if (!parameters.reconstruction_param.verbose_clock)
+    computing_timer.disable_output();
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::set_coarse_mesh_counts()
+{
+  find_vertices_positions(parameters.reconstruction_param.coarse_mesh_level);
+  calculate_counts();
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::find_all_positions()
+{
+  // Show positions in the terminal if verbose is enable
+  for (unsigned int i = 0; i < reconstruction_counts.size(); i++)
+    {
+      find_unknown_position(reconstruction_counts[i], i);
+      if (parameters.rpt_param.verbose)
+        std::cout << "Unknown particle : " << i
+                  << " Cell volume : " << cells_volumes[i]
+                  << " Position : " << reconstruction_positions[i] << std::endl;
+    }
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::find_unknown_position(
+  std::vector<double> &particle_reconstruction_counts,
+  unsigned int         id)
+{
+  int              level = 0;
+  std::vector<int> cells_indexes, new_cells_indexes;
+  Point<dim>       reconstruction_position;
+  double           volume;
+  bool             still_new_candidates = true;
+  std::string      status;
+
+  // Find the first cell candidates from coarse mesh
+  cells_indexes = find_cells(level, particle_reconstruction_counts);
+
+  // Find the positions with the more refine candidate cell
+  while (level < parameters.reconstruction_param.reactor_refinement &&
+         still_new_candidates)
+    {
+      level++; // Start at level 1
+
+      // Get vertices positions at current level and calculate their counts
+      find_vertices_positions(level, cells_indexes);
+      calculate_counts();
+
+      // Find cell candidate for the position
+      // (reconstruction counts are in the counts range of the cells for every
+      // detector)
+      new_cells_indexes =
+        find_cells(level, particle_reconstruction_counts, cells_indexes);
+
+      // If no refined cells are candidates at current level, research has to
+      // stop and need to go back to parent level
+      if (new_cells_indexes[0] == cells_indexes[0])
+        {
+          still_new_candidates = false;
+          level--;
+          status = "parent_cell"; // Cell isn't the highest refinement level
+        }
+
+      cells_indexes = new_cells_indexes;
+    }
+
+  // If many cells are candidate at the more refine level, the least squared
+  // method is used to get the best cell
+  int best_cell;
+  if (cells_indexes.size() > 1)
+    {
+      best_cell =
+        find_best_cell(level, particle_reconstruction_counts, cells_indexes);
+
+      // Update status
+      if (status.empty())
+        status = "cost_function";
+      else
+        status += "|cost_function";
```

Comment:
what? Why is the status stored as a string?  There needs to be a comment to explain this :). This is used to print afterwards right?